src/Data/List/Relation/Binary/Permutation/Propositional/Properties.agda

```diff
@@ -373,3 +375,26 @@ product-↭ (swap {xs} {ys} x y r) = begin
   (y * x) * product ys ≡⟨ *-assoc y x (product ys) ⟩
   y * (x * product ys) ∎
   where open ≡-Reasoning
+
+------------------------------------------------------------------------
+-- catMaybes
+
+catMaybes-↭ : xs ↭ ys → catMaybes xs ↭ catMaybes ys
+catMaybes-↭ refl = refl
+catMaybes-↭ (trans xs↭ ↭ys) = trans (catMaybes-↭ xs↭) (catMaybes-↭ ↭ys)
+catMaybes-↭ {xs = .mx ∷ xs} {.mx ∷ ys} (prep mx xs↭)
+  with ih ← catMaybes-↭ xs↭ | mx
+... | nothing = ih
+... | just x  = prep x ih
+catMaybes-↭ {xs = .mx ∷ .my ∷ xs} {.my ∷ .mx ∷ ys} (swap mx my xs↭)
```

Comment:
In light of the `Setoid.Properties.Maybe` versions below, is there any longer any need to expose the forced implicit bindings for `xs` and `ys`? I'd reckon probably not...?